core/src/main/scala/cats/syntax/monad.scala

```diff
@@ -20,3 +24,28 @@ final class MonadIdOps[A](private val a: A) extends AnyVal {
    */
   def iterateUntilM[F[_]](f: A => F[A])(p: A => Boolean)(implicit M: Monad[F]): F[A] = M.iterateUntilM(a)(f)(p)
 }
+
+final class MonadBooleanOps[F[_]](private val fa: F[Boolean]) extends AnyVal {
+
+  /**
+   * Behaves like `&&` but inside the `F` context.
+   *
+   * Wont evaluate `other` unless this evaluates to `true`
+   */
+  def andM(other: => F[Boolean])(implicit F: Monad[F]): F[Boolean] =
+    F.flatMap(fa) {
+      case true  => other
+      case false => F.pure(false)
+    }
+
+  /**
+   * Behaves like `||` but inside the `F` context.
+   *
+   * Wont evaluate `other` unless this evaluates to `false`
+   */
+  def orM(other: => F[Boolean])(implicit F: Monad[F]): F[Boolean] =
+    F.flatMap(fa) {
+      case true  => F.pure(false)
+      case false => other
+    }
+}
```

Comment:
This only needs Apply I think, not Monad.

Reply:
Well, not. That is exactly the point of this addition.

Right now people can just do `(fa, fb).mapN(_ && _)` but that will evaluate `fb` even when it is not needed, if you need to preserve that _"laziness"_ then you to do this exactly `flatMap` which, IMHO, hides the intention.

Reply:
That makes it no longer short-circuit which is the same kind of footgun that ifA has

Reply:
My comment got moved. This comment was on xor but my review and your push crossed paths.

XOR only needs Apply. I understand flatMap on the others.

Reply:
@johnynek ah right, my bad, sorry.
Yeah, I removed `xor` exactly for that.